realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -31,12 +31,45 @@
  * In Realm you define your model classes by sub-classing RealmObject and adding fields to be
  * persisted. You then create your objects within a Realm, and use your custom subclasses instead
  * of using the RealmObject class directly.
- * <br>
+ *
  * An annotation processor will create a proxy class for your RealmObject subclass. The getters and
  * setters should not contain any custom code of logic as they are overridden as part of the annotation
  * process.
- * <br>
+ *
+ * A RealmObject is currently limited to the following:
+ *
+ * - Private fields.
+ * - Getter and setters for these fields.
+ * - Static methods.
+ *
+ * The following field data types are supported (No boxed types):
```

Comment:
We should probably document how stuff is represented internally in Realm somewhere. I don't think is the right place as from a user standpoint it don't matter (ie. if you have a short, it will be a short going in and a coming going out).